server/src/main/java/org/elasticsearch/action/search/SearchResponse.java

```diff
@@ -200,6 +310,47 @@ public SearchResponse(
         Clusters clusters,
         String pointInTimeId
     ) {
+        return new SearchResponse(
+            hits,
+            aggregations,
+            suggest,
+            timedOut,
+            terminatedEarly,
+            profileResults,
+            numReducePhases,
+            scrollId,
+            totalShards,
+            successfulShards,
+            skippedShards,
+            tookInMillis,
+            shardFailures,
+            clusters,
+            pointInTimeId,
+            null
+        );
+    }
+
+    protected SearchResponse(
+        SearchHits hits,
+        Aggregations aggregations,
+        Suggest suggest,
+        boolean timedOut,
+        Boolean terminatedEarly,
+        SearchProfileResults profileResults,
+        int numReducePhases,
+        String scrollId,
+        int totalShards,
+        int successfulShards,
+        int skippedShards,
+        long tookInMillis,
+        ShardSearchFailure[] shardFailures,
+        Clusters clusters,
+        String pointInTimeId,
+        SearchResponseMetrics searchResponseMetrics
+    ) {
+        if (searchResponseMetrics != null) {
+            searchResponseMetrics.recordTookTime(tookInMillis);
+        }
```

Comment:
wouldn't it be nice to remove this null invariant? Could we not record the took time already in the static method that calls this? Then we don't have to pass it to the constructor either?